rwengine/src/objects/VehicleObject.cpp

```diff
@@ -161,47 +118,78 @@ void VehicleObject::tickPhysics(float dt)
 {
 	RW_UNUSED(dt);
 
-	if( physVehicle )
+	dynamics.tickPhysics(dt);
+
+	if( collision->getBulletBody() )
 	{
-		// todo: a real engine function
-		float velFac = info->handling.maxVelocity;
-		float engineForce = info->handling.acceleration * throttle * velFac;
-		if( fabs(engineForce) >= 0.001f ) physBody->activate(true);
+		float currentVelocity = (getVelocity()*60*60)/1000.f;
+		float velFac = 1.f;
+		if (currentVelocity >= info->handling.maxVelocity)
+		{
+			velFac = 0.f;
+		}
```

Comment:
Should probably add a comment (@todo) here with a [link to cerbera who figured out that the max velocity is actually mostly depending on the model shape + acceleration instead.](http://projectcerbera.com/gta/3-vc/tutorials/acceleration)